splitter.go

```diff
@@ -69,6 +66,25 @@ func (r SplitReader) ReadTag(name string, value interface{}) error {
 	return err
 }
 
+func (r SplitReader) readValue(name string, val reflect.Value) error {
+	if !val.CanAddr() {
+		return fmt.Errorf("Cannot address %s", name)
+	}
+
+	valPointer := val.Addr().Interface()
+	if val.Kind() == reflect.Ptr {
+		// Since val actually is a pointer, we want its value instead.
+		if val.IsNil() {
+			// It's currently a nil pointer, so we need to allocate and set the value
+			newVal := reflect.New(val.Type().Elem())
```

Comment:
Just a question for my own sake: is this essentially allocating a zero valued `val.Type()`?

Reply:
Immediately after I use `reflect`, I forget how it works. 😆

Short answer: approximately yes.

Medium answer: in this case, `val.Type()` is `*someType`, so we need `val.Type.Elem()` to get `someType`, and that's what we allocate.

Long answer:

The point of this code is to allow a user to have a pointer-to-value in their struct and we'll fill that. Otherwise they'd have to pre-allocate all of their pointers before calling this code, which is just irritating.

The input to `readValue` is a struct field. (Note it's the `reflect.Value` for the field, _not_ a pointer to the field). If it turns out that field is a pointer and it's `nil`, then I get the `reflect.Type` of the thing we want to point to, and `reflect.New` allocates a new one. We then set the struct's field pointer to use that value, which gets filled in a normal call to `ReadTag`.